src/System.Text.Utf8/System/Text/Utf8/Utf8String.cs

```diff
@@ -78,6 +78,11 @@ public Utf8String(RuntimeFieldHandle utf8Data, int length) : this(CreateArrayFro
         {
         }
 
+        public static explicit operator Utf8String(ArraySegment<byte> arraySegment)
```

Comment:
I have been trying to call such parameters "utf8Bytes" to communicate to callers that they need to ensure that the bytes are actually valid UTF8.